src/pipecat/utils/text/markdown_text_filter.py

```diff
@@ -73,12 +86,127 @@ def filter(self, text: str) -> str:
             filtered_text = re.sub(r"\|", "", filtered_text)
             filtered_text = re.sub(r"^\s*[-:]+\s*$", "", filtered_text, flags=re.MULTILINE)
 
+            # Remove code blocks
+            if self._settings.filter_code:
+                filtered_text = self.remove_code_blocks(filtered_text)
+
             # Restore numbered list items
             filtered_text = filtered_text.replace("§NUM§", "")
 
             # Restore leading and trailing spaces
             filtered_text = re.sub("§", " ", filtered_text)
 
-            return filtered_text
+            return filtered_text or " "  # Return a space to satisfy TTS services
```

Comment:
I'm not sure if there's a safer way to do this, but Cartesia always wants some text when a message is received. The space is a noop and satisfies the requirement.

Reply:
If it's a Cartesia requirement then it should go in Cartesia not here. So we would just return `filtered_text`.

Reply:
Thanks @aconchillo. Good call. I fixed it. Updates on this are in the description.